docsource/use_cases/module_renaming.rst

```diff
@@ -0,0 +1,169 @@
+Module Renaming
++++++++++++++++
+
+Example
+-------
+
+From version 14.0 to version 15.0, the ``crm_iap_lead_enrich`` module
+has been renamed into ``crm_iap_enrich``.
+
+See `Full v14 Module <https://github.com/odoo/odoo/tree/14.0/addons/crm_iap_lead_enrich>`_.
+
+See `Full v15 Module <https://github.com/odoo/odoo/tree/15.0/addons/crm_iap_enrich>`_.
+
+
+Result without migration script / Expected Result
+-------------------------------------------------
+
+V14 table ``ir_module_module``
+""""""""""""""""""""""""""""""
+
+.. csv-table::
+   :header: "id", "name", "state", "latest_version"
+
+   "200", "crm_iap_lead_enrich", "installed", "14.0.1.1"
+
+V15 table ``ir_module_module`` (Without openupgrade)
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+.. csv-table::
+   :header: "id", "name", "state", "latest_version"
+
+   "200", "crm_iap_lead_enrich", "to upgrade", "14.0.1.1"
+   "405", "crm_iap_enrich", "uninstalled", ""
+
+
+**Problem**:
+
+- The v14 module remains in a ``to upgrade`` state.
+- The v15 module has not been installed.
+
+As a result, the models of the module have not been loaded, the related features
+are not available, and the database is in a unstable state.
+
+V12 table ``ir_module_module`` (With openupgrade)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+.. csv-table::
+   :header: "id", "create_date", "write_date"
+
+   "200", "crm_iap_enrich", "installed", "15.0.1.1"
+
+Contribution to OpenUpgrade
+---------------------------
+
+apriori.py file
+"""""""""""""""
+
+The list of the renamed module is present in a ``apriori.py`` file in each version
+of openupgrade.
+
+* Until version 13, the file is present in this path:
+  ``odoo/addons/openupgrade_records/lib/apriori.py``.
+
+* from version 14.0 onwards, the file is present in this path:
+  ``openupgrade_scripts/apriori.py``.
+
+Add a key in the following dict:
+
+.. code-block:: python
+
+    renamed_modules = {
+        "crm_iap_lead_enrich": "crm_iap_enrich",
+    }
+
+See `Full v15 apriori file <https://github.com/OCA/OpenUpgrade/blob/97491cb7d9a8ed494a49cf1db9b7fc8852aac254/openupgrade_scripts/apriori.py#L9-L27>`_.
+
+pre-migration.py file
+"""""""""""""""""""""
+
+in the migration folder of the ``base`` module, add:
+
+.. code-block:: python
+
+    from odoo.addons.openupgrade_scripts.apriori import renamed_modules
+
+    @openupgrade.migrate(use_env=False)
+    def migrate(cr, version):
+        openupgrade.update_module_names(cr, renamed_modules.items())
+
+See `Full v15 base pre migration file <https://github.com/OCA/OpenUpgrade/blob/97491cb7d9a8ed494a49cf1db9b7fc8852aac254/openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py#L73>`_.
+
+Note: The addition is done at the initialization of the OpenUpgrade Project,
+for each new version. You will not have to do this,
+if you are working on a version of OpenUpgrade under development.
```

Comment:
I find that the note is not clear. What do you mean by a version of "OpenUpgrade under development" ?

Reply:
I mean that when openupgrade maintainers are initializing the repo for a new version (16.0 for example), they populate the apriori.py file, so, after that, there is no need to add new entries for odoo modules. (theoretically!).
but thinking about that point, maybe it's not allways true. So I propose to remove this note. What do you think ?

Reply:
I will then suggest this
```suggestion
Note: The addition is done at the initialization of the OpenUpgrade Project,
for each new version. So you will generally not have to do this, as it's only OpenUpgrade maintainers that initialize the OpenUpgrade Project.
```
Ore something like this. :)